packages/@lwc/engine/src/framework/observable-fields.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { ComponentConstructor, ComponentInterface } from './component';
+import { getComponentVM } from './vm';
+import assert from '../shared/assert';
+import { valueMutated, valueObserved } from '../libs/mutation-tracker';
+import { isRendering, vmBeingRendered } from './invoker';
+import { defineProperty, isFalse } from '../shared/language';
+
+export function observeFields(Ctor: ComponentConstructor, fields: string[] | undefined) {
+    if (fields) {
```

Comment:
`!isUndefined(fields)`

in fact I think that this condition should happens in the caller of this method... and this method, if invoked, should always have at least one field.

Reply:
this method should also receive the proto, instead of the Ctor, that's easier to reason about... you pass the obj that receive the descriptors... alternative, you call this method with the field list... and it returns a `PropertyDescriptorMap` that then you install on any obj, I like that better, it disconnects this from the ctor completely.

Reply:
you can rename this method to `createObservableFieldsMap(fields: string[])`